src/jenkins/TestManifest.groovy

```diff
@@ -10,30 +10,64 @@
 package jenkins
 
 class TestManifest {
+
     class Ci implements Serializable {
         class Image implements Serializable {
+
             String name
             String args
 
             Image(Map data) {
                 this.name = data.name
                 this.args = data.args
             }
+
         }
 
         Image image
 
         Ci(Map data) {
             this.image = new TestManifest.Ci.Image(data.image)
         }
+
+    }
+
+    class Components extends HashMap<String, Component> {
+
+        Components(ArrayList data) {
+            data.each { item ->
+                Component component = new Component(item)
+                this[component.name] = component
+            }
+        }
+
+    }
+
+    class Component implements Serializable {
+
+        String name
+
+        Component(Map data) {
+            this.name = data.name
+        }
+
     }
 
     String name
 
     Ci ci
+    Components components
 
     TestManifest(Map data) {
         this.name = data.name
         this.ci = data.ci ? new TestManifest.Ci(data.ci) : null
+        this.components = data.components ? new TestManifest.Components(data.components) : null
```

Comment:
A bit confusion.
Isnt data.components is already the components from TestManifest?

Reply:
No it was only present in buildManifest class not the Test Manifest class.

Reply:
Ok got it.